### PR TITLE
feat(updater): Report task statuses and task errors

### PR DESCRIPTION
**Pull request by BaptisteFoy**
### What does this PR do?
Reports task statuses & task errors to the RC/FA backend

### Motivation
- We want to display update errors in the UI for customers. To do so, we have to report task errors in a friendly way, with error details if needed. To do that, we add a code that will match a nice UI description, and we'll add the error content in a "details" section (like in logs)
- We also need to properly report the task status to the backend

### Additional Notes
N/A

### Possible Drawbacks / Trade-offs
N/A

### Describe how to test/QA your changes
- Launch the updater & trigger a task
- Checks that the status is properly reported in the next RC payload
- Trigger a task with an error (wrong hash in the download for instance)
- Check that the task error is sent in the next RC payload

### CODE REVIEW

**Review by ameske — APPROVED** [review]

**Review by coignetp — APPROVED** [review]

File: pkg/updater/updater.go [hunk 1]
```diff
@@ -257,23 +258,85 @@ func (u *updaterImpl) handleRemoteAPIRequest(request remoteAPIRequest) error {
 		if err != nil {
 			return fmt.Errorf("could not unmarshal start experiment params: %w", err)
 		}
-		return u.StartExperiment(context.Background(), params.Version)
+
+		u.setPackagesStateTaskRunning(request.ID)
+		var taskErr error
+		defer func() {
```
Comment (arbll) [inline pkg/updater/updater.go:264]: nit: no defer

File: pkg/updater/local_api.go [hunk 2]
```diff
@@ -136,7 +137,9 @@ func (l *localAPIImpl) startExperiment(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	log.Infof("Received local request to start experiment for package %s version %s", pkg, request.Version)
-	err = l.updater.StartExperiment(r.Context(), pkg, request.Version)
+
```
Comment (arbll) [inline pkg/updater/local_api.go:140]:
```suggestion
```

File: pkg/updater/updater.go [hunk 3]
```diff
@@ -46,9 +47,9 @@ type Updater interface {
 	Stop(ctx context.Context) error
 
 	Bootstrap(ctx context.Context, pkg string) error
-	StartExperiment(ctx context.Context, pkg string, version string) error
-	StopExperiment(pkg string) error
-	PromoteExperiment(pkg string) error
+	StartExperiment(ctx context.Context, pkg string, taskID string, version string) error
```
Comment (arbll) [inline pkg/updater/updater.go:50]: Nit: `taskID string` should be after version

**Review by arbll — APPROVED** [review]
Two micro-nits, feel free to merge after that

**Pull request merged**